spec/models/challenge_assignment_spec.rb

```diff
@@ -83,4 +83,75 @@
     end
   end
 
+  describe "request_signup_pseud=" do
+    let!(:collection) { create(:collection, challenge: create(:gift_exchange)) }
+
+    let(:assignment) { collection.assignments.build }
+
+    context "when a user has signed up with a pseud matching their login" do
+      let(:user) { create(:user) }
+
+      let!(:signup) do
+        create(:challenge_signup,
+               collection: collection,
+               pseud: user.default_pseud)
+      end
+
+      it "assigns the user's signup when entering the user's login" do
+        assignment.request_signup_pseud = user.login
+        expect(assignment.request_signup).to eq(signup)
+      end
+
+      context "when another user has signed up with the same pseud name" do
+        let(:ambiguous) { create(:pseud, name: user.login) }
+
+        let(:ambiguous_signup) do
+          create(:challenge_signup,
+                 collection: collection,
+                 pseud: ambiguous_signup)
+        end
+
+        it "assigns the first user's signup when entering the first user's login" do
```

Comment:
Should we also test the flip side, i.e., it assigns the second user's signup when entering `ambiguous.name (ambiguous.user.login)`?

Reply:
I added an extra test (and in doing so discovered and fixed a bug with the existing test).

Do you want me to do it for the `OwnedTagSet` tests, as well?

Reply:
Yay! And yes please, sorry for not being clear.